t/Network.t

```diff
@@ -369,39 +370,46 @@ use Net::Works::Network;
 
     is(
         $net->as_string(), '0.0.0.0/32',
-        'a network created via new_from_integer with a uint128 integer and version => 4 stringifies correctly'
+        'a network created via new_from_integer with a '
+            . ref $int
+            . ' integer and version => 4 stringifies correctly'
     );
 }
 
-{
+for my $two ( uint128(2), Math::BigInt->new(2) ) {
+    my $type = ref $two;
```

Comment:
Yeah, that might make sense. There is also a story for making the `new_from_integer` behave better. The test changes in this branch might be useful for that as ideally it would accept any large number type as long as that type stringified correctly.